lib/pg.ddl.js

```diff
@@ -293,62 +374,63 @@ const generateTable = (
   } = categorizeEntries(category, name, categories);
 
   const maxPropLength = getMaxPropLength(properties);
-  const resolvableLinks = unresolvedLinks.get(name) || [];
-  const unresolved = links.filter(({ name: propName, property: link }) => {
-    if (link.category === name || existingTables.includes(link.category)) {
-      resolvableLinks.push({
-        from: name,
-        to: link.category,
-        name: propName,
-        link,
-      });
-      return false;
-    }
-    return true;
-  });
-
   const props = generateProperties(name, properties, types, maxPropLength);
   props.unshift(generateId(type, maxPropLength));
 
-  const primaryKey =
-    `\nALTER TABLE ${name} ADD CONSTRAINT pk${name}Id PRIMARY KEY (Id);\n`;
-
-  let linksSQL = generateLinks(resolvableLinks, categories);
-  if (linksSQL) {
-    linksSQL = `\n${linksSQL}\n`;
-  }
+  const [unresolved, resolvableLinks] =
+    getResolvableLinks(category, unresolvedLinks, links, existingTables);
```

Comment:
So, `category` is an object here, according to `generateTable` method's documentation, but `getResolvableLinks` expects a string as a first argument, according to its documentation.
This leads to a `TypeError` being thrown in the `requiresForeignKey` function later, and thus at the moment, the generator doesn't seem to be working.